### ISSUE TITLE
[BUG] EventHub .NET Sample. Wrong instructions in README

### ISSUE THREAD

**Issue by rido-min** [Azure/azure-sdk-for-net#17263]
Following https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample01_HelloWorld.md I found missing instructions:

**Expected behavior**
I can follow and complete the tutorial

**Actual behavior (include Exception or Stack Trace)**
The sample code does not compile

I fixed by following the next steps.

1. Add main NuGet `dotnet add package Azure.Messaging.EventHubs`
2. Update AMQP to workaround: https://github.com/Azure/azure-amqp/issues/165 
3. Add `dotnet add package System.Memory.Data` to be able to use `BinaryData`
4. Update code to avoid using `partitionEvent.Data.EventBody`

I update the code from 

```csharp
await foreach (PartitionEvent partitionEvent in consumer.ReadEventsAsync(cancellationSource.Token))
{
    eventDataRead.Add(partitionEvent.Data.EventBody.ToString());
    if (eventDataRead.Count >= maximumEvents)
    {
        break;
    }
}
```
to:

```csharp

await foreach (PartitionEvent partitionEvent in consumer.ReadEventsAsync(cancellationSource.Token))
{
    var data = BinaryData.FromBytes(partitionEvent.Data.Body);
    eventDataRead.Add(data.ToString());
                    
    if (eventDataRead.Count >= maximumEvents)
    {
        break;
    }
}
```

**Comment by jsquire** [comment]
Hi @rido-min.  Thank you for your feedback.   Looks like we missed updating the samples README to reference the 5.3.0-beta.4 package.   I'll get that patched up.

**Comment by jsquire** [comment]
Fixed by #17315

**Comment by devedse** [comment]
This issue also happens on other documentation pages like:
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample05_ReadingEvents.md
